agent.py

```diff
@@ -176,18 +175,22 @@ def __init__(self, agent, mib, cfg):
         self._cfg = cfg
         self.setDaemon(True)
 
-        self.client = QumuloClient(cfg.clusters[0]) # only one cluster for now
+        self.client = QumuloClient(cfg.clusters[0])  # only one cluster for now
         self.notified_offline = False
         self.notified_dead_drives = False
-        self.notified_power_supply_failure = False
+        # Use an array of dictionaries to track per-node PS notify states
+        self.notified_power_supply_failure = \
+            [{'PS1': False, 'PS2': False}
```

Comment:
I prefer  [{'PS1': False, 'PS2': False}]*len(cfg.clusters[0].ipmi.ipmi_servers) but I'm also not an engineer and I wouldn't no a best practice if it knocked me on my ass. :-)